basket/Basket.js

```diff
@@ -813,6 +837,40 @@ export default class Basket {
     return true
   }
 
+  /**
+   * Is the booked time span within the available dates of a product (considering availability range and validity dates)
+   * @param basketEntries: BasketEntry[]
+   * @return {boolean}
+   */
+  checkAvailableDates(basketEntries) {
+    if (!basketEntries || !basketEntries.length) {
+      throw new Error('No product definitions provided in duration check')
+    }
+    const shopModulesInstance = store.getters.getShopModulesInstance()
+    // iterate product definitions
+    for (let i = 0; i < basketEntries.length; i++) {
+      const basketEntry = basketEntries[i]
+      if (basketEntry.isRequiredEntry() || basketEntry.isEventEntry()) {
+        // no product would be found
+        continue
+      }
+      const bookingStart = basketEntry.getValidFrom()
+      const productDefinition = basketEntry.getProductDefinition()
+      const productId = productDefinition.getProductId()
+      const productInstance = shopModulesInstance.getProductInstanceByProductId(
+        productId
+      )
+      const checkResult = productInstance.checkAvailableDates(
+        productDefinition,
+        bookingStart
+      )
+      if (!checkResult) {
+        return false
+      }
```

Comment:
this will exit the loop - is this correct?

Reply:
yeah. correct. intended.